### ISSUE TITLE
First screen for Simplex flow

### ISSUE THREAD

**Issue by bdresser** [MetaMask/Design#90]
We're integrating with Simplex for crypto purchases via credit card.

While most of the flow is an iFrame from simplex that we don't control, we CAN design the first step of the flow. It asks the user for their ETH address, and amount of ETH <> amount of fiat. Both the latter fields are editable and the page will convert between the two. All other styles are totally up to us. 

One from Abra looks like this:

![image.png](https://images.zenhubusercontent.com/5b0f0677c7faeb213bec8e7c/aa65aa6a-b153-4e5b-a46a-630589b9a4b4)

Some notes

- Still unclear if it's technically possible for this to live in the extension, or if it has to live on metamask.io - will know for sure once Simplex responds.
- Let's default the fiat currency to whatever the user has selected in the extension. If we're linking to metamask.io, we can include this as a query paramete
- We will be able to inject the address in to this form by hard-coding the html field value. May be nice to let the user know this is "Account 1" or something like that. Let's still allow the address to be editable, though, in case the user wants to send elsewhere.

**Comment by omnat** [comment]
<img width="1626" alt="Screen Shot 2019-05-01 at 3 38 10 PM" src="https://user-images.githubusercontent.com/44038140/58131567-db5afd00-7bd3-11e9-89bc-066618dd6e86.png">

Dapper has their name still there - this is good to let user know this is still in context of Metamask and is legit linking. Suggest to show MetaMask branding somewhere.

**Comment by bdresser** [comment]
Also will need a simple ending page for the flow 

> All set! Your ETH will arrive in your MetaMask wallet shortly.
Return to MetaMask at any time by clicking the Fox on your menu bar
[ arrow / foxy icon / point up]

**Comment by cjeria** [comment]
Here's a first pass design for this page. I looked at [Abra's page](https://www.abra.com/buybitcoin/simplex/?fiat=USD&ref_id=100&link=3ihRYNZbgO) and it looks like everything surrounding the exchange widget is customizable. We'd ideally use the graphic motif that the illustrator comes up with (inspired by this image) here. 

Also, the widget seems to be responsive so I'd be surprised if we couldn't embed it inside the extension, although full page would be a nicer experience IMO.

Feedback welcomed!

![simplex MM page](https://user-images.githubusercontent.com/597770/59798896-6b0dcd00-92a0-11e9-945b-334a47623cbe.png)

**Comment by cjeria** [comment]
@bdresser can the widget itself be customized? 🤔

**Comment by bdresser** [comment]
@cjeria yes the widget can be customized! We can plop those form inputs right on the page, don't need to style the box like they did. Should be pretty simple. 

AFTER the user leaves this page, they will be in a simplex-styled experience, and we can have our logo on the top.

**Comment by omnat** [comment]
In reference to chat on design sync. Is pre-filling current metamask account address in this input field possible? 
cc @cjeria 

<img width="942" alt="Screen Shot 2019-06-05 at 5 34 16 PM" src="https://user-images.githubusercontent.com/44038140/60045407-aaaa2f80-9679-11e9-9757-f0b46c83b247.png">

**Comment by bdresser** [comment]
@omnat yes

**Comment by cjeria** [comment]
- Update graphic motif to the original version
- Update form inputs per our design system

**Comment by omnat** [comment]
- Prefill current Metamask account of user
- Account switching within the form

**Comment by cjeria** [comment]
@danjm just to confirm prefilling the `to` field is possible in the Simplex page?
Also, is account switching also possible in this screen?

**Comment by omnat** [comment]
@cjeria communicate to kyokan the final on this

**Comment by cjeria** [comment]
Here's a new design for the simplex page. I also received confirmation from @danjm that account switching within this page will not be possible. 

I did a lil competitive analysis and referenced Dapper's version of their simplex page. I took some notes on a screenshot of their page (attached below) and had some additional questions.

Here's the latest design iteration: [Figma](https://www.figma.com/file/TTsx3ZRbuW07PFhdCqq43cck/Simplex-page?node-id=17%3A0)
![image](https://user-images.githubusercontent.com/597770/64738268-c5dc4c80-d4bc-11e9-981f-6a9e935eaade.png)

Questions after looking at Dapper's Simplex page:
1 - There’s no Simplex logo on this screen.. Can we also get away with not showing it here as well? Is a link to simplex enough?
2- Should we display the Simplex service fee?
3 - If we add the service fee details an info tooltip is nice (see Dappers screenshot). Any opposition to adding one?
4 - Nice prep copy for setting expectations below the inputs "Note: you'll need to provide a piece of government-issued ID to complete your purchase. Please be ready with either your passport, driver's license, or ID card."

@bdresser when you get a chance, can you take a look?

![image](https://user-images.githubusercontent.com/597770/64738226-ac3b0500-d4bc-11e9-9ecd-3e954a2d23b0.png)

**Comment by bdresser** [comment]
>1 - There’s no Simplex logo on this screen.. Can we also get away with not showing it here as well? Is a link to simplex enough?

they've told me the header is mandatory. not sure the details of how this integration is built; i can ask the simplex team if you'd like
>2- Should we display the Simplex service fee?
>3 - If we add the service fee details an info tooltip is nice (see Dappers screenshot). Any opposition to adding one?

sure thing. our details are slightly different - i believe it's 3.5% simplex, 1.5% us, but jason can confirm. i would be cool with a generalized text that says "there's a 5% fee" 

>4 - Nice prep copy for setting expectations below the inputs "Note: you'll need to provide a piece of government-issued ID to complete your purchase. Please be ready with either your passport, driver's license, or ID card."

yup this is nice! i think this could also live on the previous screen, where the user chooses the Simplex option -- that page is probably overdue for some design/copy love. I'm cool with including this in either place.

**Comment by cjeria** [comment]
> they've told me the header is mandatory. not sure the details of how this integration is built; i can ask the simplex team if you'd like

Once these final tweaks are made, it's possible to just show them the mockup and get any final feedback from them.

**Comment by cjeria** [comment]
Ok, so here's an updated version. Based on how I've seen other implementations of this page work (abra, dapper) the idea is that the user could insert an amount in ETH or USD and the fee will be calculated below the USD field.

The tooltip here shows a breakdown of the fee structure. I added the 1.5% line in there for transparency as our users might appreciate knowing but not opposed to removing it. 

cc @bdresser 
![image](https://user-images.githubusercontent.com/597770/64744329-60468b00-d4d1-11e9-914d-d85810e7ee58.png)

**Comment by omnat** [comment]
Do we have any feedback on this from Simplex team @bdresser ?

**Comment by bdresser** [comment]
We don't need header on the first page, but it will show once we drop the user in to the simplex iFrame. 

These look good to me!

**Comment by omnat** [comment]
Ready for development. @cjeria to hand-off in Sprint 23.
